### PR TITLE
iOS screen shot with platform view

### PR DESCRIPTION
**Pull request by cyanglaz**
Make screenshot work for iOS with embedded platform view.
The general idea is:
Added a screen_shot mode in the paint traversal.
When we are traversing the layer_tree in the screen_shot mode, we modify the platform_view_layer's paint method:
* In a regular platform_view_layer's paint, we composite the platform view, create an overlay, and all the drawing afterward will be painted on the newly created overlay.
* In a platform_view_layer's paint with screen_shot mode, we composite the platform view, take a screenshot of the platform view and convert it into an SkImage, then draw the SkImage onto the current canvas. Also, we never update the canvas to the newly created canvas from `CompositeEmbeddedView` method. So in the screen_shot mode, all the drawing happens on a single canvas regardless how many platform view exists. One optimize option would be not create the overlay uiview at all during the screen_shot mode. Since screenshot is a rare operation and the performance is not essential, I'd keep the code simple which sacrifices a little performance.

Thread merging is done on `PostPrerollAction` in the view embedder

related issue: https://github.com/flutter/flutter/issues/23435

### CODE REVIEW

File: shell/common/rasterizer.cc [hunk 1]
```diff
@@ -347,7 +348,7 @@ static sk_sp<SkData> ScreenshotLayerTreeAsImage(
                                                root_surface_transformation,
                                                false, nullptr);
   canvas->clear(SK_ColorTRANSPARENT);
-  frame->Raster(*tree, true);
+  frame->Raster(*tree, true, true);
```
Comment (cyanglaz) [inline shell/common/rasterizer.cc]: Going to merge thread before this. Haven't added the code yet.

File: shell/common/rasterizer.cc [hunk 2]
```diff
@@ -22,6 +22,8 @@ namespace flutter {
 // used within this interval.
 static constexpr std::chrono::milliseconds kSkiaCleanupExpiration(15000);
 
+static const int kDefaultScreenShotThreadMergeLeaseDuration = 10;
```
Comment (iskakaushik) [inline shell/common/rasterizer.cc]: I'd rather the rasterizer be unaware of any thread merging and keep this in view embedder.

Comment (cyanglaz) [reply shell/common/rasterizer.cc]: Updated per our offline discussion.

**Comment by amirh** [comment]
Offline discussion:

A simpler approach might be to add a
```cc
void SubmitFrameToCanvas(SkCanvas* canvas);
```
method to `ExternalViewEmbedder`, and call it from rasterizer when taking screenshots.

This way we don't have to change anything in the pipeline other than calling the new method from `rasterizer.cc` and implementing it in `FlutterPlatformViews.mm`.

**Comment by cyanglaz** [comment]
This implementation doesn't work as intended. Closing this PR

**Pull request closed without merging**